app/src/main/java/dev/andrylat/raqimbek/bankingutils/PaymentSystemDeterminer.java

```diff
@@ -1,10 +1,16 @@
 package dev.andrylat.raqimbek.bankingutils;
 
+import lombok.NonNull;
+
 import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
 
 public class PaymentSystemDeterminer {
-  Optional<PaymentSystem> determinePaymentSystemByCardNumber(String cardNumber) {
+  @NonNull
+  public Optional<PaymentSystem> determinePaymentSystem(@NonNull List<String> inputList) {
```

Comment:
Why payment system determiner should care about different inputs and so on? Why did you expect that card number will be first only? maybe tomorrow logic will be changed and card number will be second. You will need to rewrite half of app.